sphinxcontrib/needs/layout.py

```diff
@@ -829,7 +823,7 @@ def link(self, url, text=None, image_url=None, image_height=None, image_width=No
         link_node = nodes.reference(text, text, refuri=url)
 
         if image_url:
-            image_node = self.image(image_url, image_height, image_width)
+            image_node = self.image(image_url, image_height, image_width, no_link=True)
```

Comment:
@danwos I had to change this to make image links work.

Reply:
interesting. Needed some time to understand this, but I think it makes sense as there were 1-2 bugs in the code of image() and link handling links with images. Thanks.